src/app/jscaip/state/FourStatePieceGameStateWithTable.ts

```diff
@@ -19,4 +20,38 @@ export class FourStatePieceGameStateWithTable extends GameStateWithTable<FourSta
                 };
             });
     }
+
+    public isPiece(coord: Coord): boolean {
+        const piece: FourStatePiece = this.getPieceAt(coord);
+        return piece.isPlayer();
+    }
+
+    public hasPieceBelongingTo(coord: Coord, player: Player): boolean {
+        const optional: MGPOptional<FourStatePiece> = this.getOptionalPieceAt(coord);
+        if (optional.isPresent()) {
+            return optional.get().is(player);
+        } else {
+            return false;
+        }
+    }
+
+    public coordIsOccupiedSquare(coord: Coord): boolean {
```

Comment:
c'est isPiece qui retourne faux si on est dans du vide.
isOnBoardWithPiece?

Reply:
si ceux qui disent "false" quand c'est out of board méritent du "isOnBoardWith" ça va refactorer sec

Comment tu distinguerais celles qui throwent si out of board des autres :D ?

Reply:
unsafe serait ceux qui throw ? Idéalement, on évite toutes les versions qui throwent également ?

Reply:
Tu veux ticketter ça, tu proposes clairement un "une fonction de ce genre là ferais ceci-celà" etc ?
Et on MERGE ce masta tickette ?

Reply:
Tu proposes un petit nom de ticket ? "use unsafe for methods that throw" dans le rename ticket?

Reply:
c'est dans le rename ticket !